ckan/tests/controllers/test_package.py

```diff
@@ -38,6 +39,43 @@ def test_form_renders(self):
         env, response = _get_package_new_page(app)
         assert_true('dataset-edit' in response.forms)
 
+    @helpers.change_config('ckan.auth.create_unowned_dataset', 'false')
+    def test_needs_organization_but_no_organizations_has_button(self,):
+        ''' Scenario: The settings say every dataset needs an organization
+        but there are no organizations. If the user is allowed to create an
+        organization they should be prompted to do so when they try to create
+        a new dataset'''
+        app = self._get_test_app()
+        sysadmin = factories.Sysadmin()
+
+        env = {'REMOTE_USER': sysadmin['name'].encode('ascii')}
+        response = app.get(
+            url=url_for(controller='package', action='new'),
+            extra_environ=env
+        )
+        assert 'dataset-edit' not in response.forms
+        assert url_for(controller='organization', action='new') in response
+
+    @patch('ckan.logic.auth.create.organization_create',
+           return_value={'success': False})
+    @helpers.change_config('ckan.auth.create_unowned_dataset', 'false')
+    def test_needs_organization_but_no_organizations_no_button(self, *args):
+        ''' Scenario: The settings say every dataset needs an organization
+        but there are no organizations. If the user is not allowed to create an
+        organization they should be told to ask the admin but no link should be
+        presented'''
+        app = self._get_test_app()
+        sysadmin = factories.Sysadmin()
```

Comment:
When I change the test to this configuration it works. But the original problem remains. When only this test is run it works but if the previous test (`test_needs_organization_but_no_organizations_has_button`)is run before the test fails because a 403 status is returned...